packages/common-substrate/src/project/versioned/v0_2_0/model.ts

```diff
@@ -106,7 +124,7 @@ export class ProjectManifestV0_2_0Impl extends ProjectManifestBaseImpl implement
     keepDiscriminatorProperty: true,
   })
   dataSources: (RuntimeDataSourceV0_2_0 | CustomDatasourceV0_2_0)[];
-  private _deployment: DeploymentV0_2_0;
+  protected _deployment: DeploymentV0_2_0;
 
   toDeployment(): string {
```

Comment:
This function and `validate` can be moved to https://github.com/DeveloperInProgress/subql/blob/main/packages/common/src/project/versioned/base.ts#L6 and `ProjectManifestBaseImpl` can be made into an abstract class